src/circuit/blinding_circuit.rs

```diff
@@ -1,173 +1,102 @@
-use std::time::Instant;
-
+use crate::circuit::circuit_parameters::CircuitParameters;
+use crate::circuit::validity_predicate::ValidityPredicate;
+use ark_ec::twisted_edwards_extended::GroupAffine as TEGroupAffine;
+use ark_ff::UniformRand;
 use ark_poly::univariate::DensePolynomial;
 use ark_poly_commit::PolynomialCommitment;
 use plonk_core::{
+    circuit::Circuit,
     constraint_system::StandardComposer,
-    prelude::{Point, Proof},
-    proof_system::{pi::PublicInputs, Prover, Verifier},
-};
-
-use crate::circuit::{
-    circuit_parameters::CircuitParameters, gadgets::blinding::blinding_gadget,
-    validity_predicate::ValidityPredicate,
+    error::to_pc_error,
+    prelude::{Error, Point},
+    proof_system::{verifier::Verifier, Blinding, VerifierKey},
 };
+use rand::RngCore;
 
 pub struct BlindingCircuit<CP: CircuitParameters> {
-    pub public_input: PublicInputs<CP::CurveBaseField>,
-    pub proof: Proof<CP::CurveBaseField, CP::OuterCurvePC>,
-    pub verifier: Verifier<CP::CurveBaseField, CP::Curve, CP::OuterCurvePC>,
-    pub vk: <CP::OuterCurvePC as PolynomialCommitment<
-        CP::CurveBaseField,
-        DensePolynomial<CP::CurveBaseField>,
-    >>::VerifierKey,
+    pub vk: VerifierKey<CP::CurveScalarField, CP::CurvePC>,
+    pub blinding: Blinding<CP::CurveScalarField>,
+    pub zh: [CP::CurveBaseField; 2],
```

Comment:
`zh` can be obtained from the setup. Maybe we can have it outside of the `BlindingCircuit` struct ?

Reply:
Yeah, let's discuss it in my next PR, in which there are some further modifications of BlindingCircuit generation.